projects/vdk-core/src/vdk/internal/core/errors.py

```diff
@@ -13,25 +13,107 @@
 import enum
 import logging
 import re
-import sys
-import traceback
 from collections import defaultdict
 from enum import Enum
 from logging import Logger
-from types import TracebackType
-from typing import cast
 
 log = logging.getLogger(__name__)
-MSG_CONSEQUENCE_DELEGATING_TO_CALLER__LIKELY_EXECUTION_FAILURE = (
-    "I'm rethrowing this exception to my caller to process."
-    "Most likely this will result in failure of current Data Job."
-)
-MSG_CONSEQUENCE_TERMINATING_APP = (
-    "The provided Data Job will not be executed. Terminating application."
-)
-MSG_COUNTERMEASURE_FIX_PARENT_EXCEPTION = (
-    "See contents of the exception and fix the problem that causes it."
-)
+
+# ERROR TYPES
+
+
+class BaseVdkError(Exception):
+    def __init__(
+        self,
+        vdk_resolvable_actual: ResolvableByActual = None,
+        vdk_type: ResolvableBy = None,
+        *error_message_lines: str,
+    ):
+        """
+        :param vdk_resolvable_actual: who whould resolve the error (User or Platform Team)
+        :param vdk_type: the vdk error type, e.g. Platform, User or Config error
+        :param *error_message_lines: optional, the error message lines used to build the error representation
+        """
+
+        # Check if error message or dict was passed
+        # for compatibility with vdk plugins
+        error_message = ""
+        if error_message_lines and isinstance(error_message_lines[0], ErrorMessage):
+            message = error_message_lines[0]
+            error_message += message.summary
+            error_message += message.what
+            error_message += message.why
+            error_message += message.consequences
+            error_message += message.countermeasures
+        elif error_message_lines and isinstance(error_message_lines[0], dict):
+            message = error_message_lines[0]
+            error_message += "\n".join(message.values())
+        else:
+            for line in error_message_lines:
+                error_message += line
+                error_message += "\n"
+        super().__init__(error_message)
+        self._vdk_resolvable_actual = vdk_resolvable_actual
+        self._vdk_type = vdk_type
+        self._prettify_message(str(error_message))
+
+    def __str__(self):
+        return self._pretty_message
+
+    def __repr__(self):
+        return self._pretty_message
+
+    def _prettify_message(self, message):
```

Comment:
I think this method is a little bit confusing maybe adding a docstring and adding a small example would be good here since we have a lot of magic numbers and formatting

Reply:
Added docstring.